src/mpyl/reporting/targets/jira.py

```diff
@@ -70,25 +82,87 @@ class JiraConfig:
 
     @staticmethod
     def from_config(config: dict):
-        return JiraConfig(site=config['site'], user_name=config['userName'], password=config['password'],
-                          token=config.get('token'))
+        jira_config = config.get('jira')
+        if not jira_config:
+            raise KeyError('jira section needs to be defined in mpyl_config.yml')
+        return JiraConfig(site=jira_config['site'], user_name=jira_config['userName'], password=jira_config['password'],
+                          token=jira_config.get('token'))
+
+
+def create_jira_for_config(jira_config: JiraConfig):
+    return Jira(
+        url=jira_config.site,
+        token=jira_config.token,
+        api_version='2',
+        cloud=True
+    ) if jira_config.token else Jira(
+        url=jira_config.site,
+        username=jira_config.user_name,
+        password=jira_config.password,
+        api_version='2', cloud=True
```

Comment:
nit: `cloud` should be on the next line if you want to keep the formatting consistent